hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala

```diff
@@ -244,14 +236,14 @@ object Extract {
 
         val i = ab.length
         val initOps = newAggs.result()
-        val aggs = initOps.map(_.aggSig)
 
-        val rt = TDict(key.typ, TTuple(aggs.map(_.returnType.virtualType): _*))
+        val rt = TDict(key.typ, TTuple(initOps.map(_.aggSig.resultType): _*))
         newRef._typ = -rt.elementType
 
-        val aggSig = PhysicalAggSignature(Group(), Seq(PVoid), FastSeq(key.pType, PVoid), Some(aggs))
-        ab += InitOp(i, FastIndexedSeq(Begin(initOps)), aggSig)
-        seqBuilder += SeqOp(i, FastIndexedSeq(key, Begin(newSeq.result().toFastIndexedSeq)), aggSig)
+        val groupSig = AggSignature(Group(), Seq(TVoid), FastSeq(key.typ, TVoid))
```

Comment:
could we add a comment, or note in a design doc, the side-effect that providing a TVoid type signature on the agg op leads to? Seems potentially confusing to new readers of the code.

Reply:
will add a comment to this effect.